Tests/AK/TestFloatingPointParsing.cpp

```diff
@@ -577,3 +577,33 @@ TEST_CASE(invalid_hex_floats)
     EXPECT_HEX_PARSE_TO_VALUE_AND_CONSUME_CHARS("0xABCDPEF", 0xabcdp0, 6);
     EXPECT_HEX_PARSE_TO_VALUE_AND_CONSUME_CHARS("0xCAPE", 0xCAp0, 4);
 }
+
+#define BENCHMARK_DOUBLE_PARSING(value, iterations)     \
+    do {                                                \
+        auto data = #value##sv;                         \
+        auto true_result = value;                       \
+        for (int i = 0; i < iterations * 10'000; ++i) { \
+            AK::taint_for_optimizer(data);              \
+            EXPECT_EQ(data.to_double(), true_result);   \
+        }                                               \
+    } while (false)
+
+BENCHMARK_CASE(one)
```

Comment:
Is this the only benchmarks you tried?
When I wrote the code I tested on files from https://github.com/fastfloat/supplemental_test_files/tree/main/data .

I don't think we need those files in the repo, just wondering what you benchmarked on.

Reply:
Yeap, these are the only four (with the results listed above). I wasn't smart enough to google test data :) Will post results here later.
The main concern for me was small exponents and numbers 'cause original code just used cached result for `power <= 19` and I do up to 4 multiplications.